File: Data Structure/Queue/acmicpc.net/1158/Main.js
"use strict";
const inputFilePath = "/sample.txt";
const input = require("fs")
  .readFileSync(__dirname + inputFilePath)
  .toString()
  .trim()
  .split("\n");
// const input = require("fs").readFileSync("/dev/stdin").toString().trim().split("\n");
const [N, K] = input[0].split(" ").map(Number);
let ans = [];
let arr = [];
for (let i = 1; i <= N; i++) {
  arr.push(i);
}
let cursor = 0;
while (arr.length) {
  cursor += K - 1;
  if (cursor >= arr.length) cursor %= arr.length;
  ans.push(arr.splice(cursor, 1));
}

console.log("<" + ans.join(", ") + ">");

File: Data Structure/Queue/acmicpc.net/1158/Main.py
from collections import deque
import sys
# input = sys.stdin.readline
file = open("./01 ing/acmicpc.net/sample.txt", "r")
input = file.readline
n, m = map(int, input().split())
ans = []
q = deque()
for i in range(n):
    q.appendleft(i+1)

while len(q) != 0:
    for item in range(m):
        if(item == m-1):
            ans.append(str(q.pop()))
        else:
            q.appendleft(q.pop())

print("<" + ", ".join(ans) + ">")
